### PR TITLE
px4_ros_com overhaul corresponding with new PX4-Autopilot microdds_client

### PR DESCRIPTION
**Pull request by dagar**
- https://github.com/PX4/PX4-Autopilot/pull/20227
 - px4_ros_com is now simply a handful of examples, no more agent or generation required
 - micro_ros_agent started automatically in launch examples

### CODE REVIEW

File: launch/offboard_control_launch.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+launch:
+
```
Comment (spiderkeys) [inline launch/offboard_control_launch.yaml:2]: Is this meant to launch standalone, or also automatically launch the micro_dds_agent, similar to sensors_combined_listener launchfile?

File: scripts/build_ros2_workspace.bash [hunk 2]
```diff
@@ -5,7 +5,6 @@ set -e
 if [[ $1 == "-h" ]] || [[ $1 == "--help" ]]; then
   echo -e "Usage: build_ros2_workspace.bash [option...] \t This script builds px4_ros_com workspace for ROS 2" >&2
   echo
-  echo -e "\t--add_ros1_bridge \t Clone ros1_bridge. To be used in a further build in case a ROS1 workspace is used. Check 'build_all' or 'build_ros1_bridge' scripts."
   echo -e "\t--ros_distro \t\t Set ROS 2 distro name (dashing|eloquent|foxy|galactic|humble|rolling). If not set, the script will set the ROS_DISTRO env variable based on the Ubuntu codename"
```
Comment (spiderkeys) [inline scripts/build_ros2_workspace.bash:8]: Wondering if everything here actually still works (or there is an intent to make work) with pre-Foxy versions. Might save some future headaches to draw a line in the sand here for which distros are expected/tested to work with PX4.

Comment (dagar) [reply scripts/build_ros2_workspace.bash:8]: I was personally planning to draw the line at Foxy, but we can be flexible if there truly is a compelling reason.

File: launch/offboard_control_launch.yaml (hunk 1 above)
Comment (dagar) [reply launch/offboard_control_launch.yaml:2]: Yes, I was just playing around so far to get a feel for python vs yaml launch. Do you have a preference?

**Comment by swimmingseeds** [comment]
This pull request has been mentioned on **Discussion Forum for PX4, Pixhawk, QGroundControl, MAVSDK, MAVLink**. There might be relevant details there:

https://discuss.px4.io/t/px4-to-ros2-connection-does-not-work/31275/4

**Pull request merged**